internal/cmd/test/test.go

```diff
@@ -1,232 +1,46 @@
 package test
 
 import (
-	"bytes"
+	_ "embed"
+
 	"fmt"
-	"io"
 	"os"
-	"os/exec"
-	"path/filepath"
 	"strings"
 
-	"github.com/gobuffalo/envy"
-	"github.com/gobuffalo/meta"
-	"github.com/gobuffalo/pop/v6"
-	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
-func runE(c *cobra.Command, args []string) error {
-	os.Setenv("GO_ENV", "test")
-	if _, err := os.Stat("database.yml"); err == nil {
-		// there's a database
-		test, err := pop.Connect("test")
-		if err != nil {
-			return err
-		}
-
-		// drop the test db:
-		if err := test.Dialect.DropDB(); err != nil {
-			// not an error, since the database will be created in the next step anyway
-			logrus.Info("no test database to drop")
-		}
-
-		// create the test db:
-		err = test.Dialect.CreateDB()
-		if err != nil {
-			return err
-		}
-
-		// Read and remove --force-migrations flag from args:
-		forceMigrations := strings.Contains(strings.Join(args, ""), "--force-migrations")
-		args = cutArg("--force-migrations", args)
-		if forceMigrations {
-			fm, err := pop.NewFileMigrator("./migrations", test)
-			if err != nil {
+var (
+	//go:embed longhelp.txt
+	longHelp string
+	Cmd      = &cobra.Command{
+		Use:   "test",
+		Short: "Runs tests for your Buffalo app",
+		Long:  longHelp,
+
+		// DisableFlagParsing is set to true since we will need to allow undefined
+		// flags to be passed to the go test command.
+		DisableFlagParsing: true,
```

Comment:
Actually, I would like to change this behavior.

Reply:
Cobra does not support changing this behavior. AFAIK if we enable this we will have to  listen for each of the flags that the go build and test command accept, otherwise Cobra will err.

Reply:
I explained more clearly (with my limited Yonglish skill) in the issue #241. In short, We can use "end-of-options" for this. Cobra supports the POSIX standard "end-of-options" flag which is the double dash `--`, and using it could be acceptable for most Unix-like OS users naturally. By using this, we can both use Cobra's flag parsing for Buffalo-specific flags and pass others to `go test` very easily.

Reply:
Yeah, the end-of-options is the POSIX standard and it will make us and users easy to separate buffalo-specific flags such as `--force-migrations` and `go test` flags including app-specific flags with the standard flag `-args`. However, I am still open to a simpler way too:

**POSIX style** with simpler internal with Cobra (Cobra makes the args for the `go test`)

```console
$ buffalo test --force-migration -- -cover -race ./...
```

vs.

**Go style** without end-of-options but with slightly more work of cleaning buffalo-specific args from the original args

```console
$ buffalo test --force-migration -cover -race ./...
```